architecture/src/commonMain/kotlin/lifecycle/LifecycleSubscribableMarker.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ Copyright 2021 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.architecture.lifecycle
+
+import com.splendo.kaluga.architecture.viewmodel.ViewModel
+
+/**
+ * Implementing this interface as a public property of a [ViewModel] allows for automatic binding to platform specific lifecycle aware manager.
+ *
+ * Be aware that classes implementing this interface that reside on a ViewModel as a property will have their getter invoked on instantiating the ViewModel to check if the instance is lifecycle aware (if the platform supports this).
+ * This means, for example, that `lazy` delegation will have little effect.
```

Comment:
That was the case, but it was fixed in scope of #257 .

Reply:
No, that implementation still [invokes the getter](https://github.com/splendo/kaluga/pull/257/files#diff-b4751f068b2271856789111619ebceb23ed24e17eb4c07024c1d1be0fca2ba9aR57) but only (as written here) if this interface is detected first.

I feel that's ok as long as you are warned about it, which I add in this commit :)

Reply:
Ah, I see, yes, you're right, I misunderstood the comment.